specs/phase1/data-availability-sampling.md

```diff
@@ -0,0 +1,150 @@
+# Ethereum 2.0 Phase 1 -- Data Availability Sampling
+
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+## Table of contents
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Custom types
+
+We define the following Python custom types for type hinting and readability:
+
+| Name | SSZ equivalent | Description |
+| - | - | - |
+| `SampleIndex` | `uint64` | A sample index, corresponding to chunk of extended data |
+| `BLSPoint` | `uint256` | A number `x` in the range `0 <= x < MODULUS` |
+
+
+## New containers
+
+### DASSample
+
+```python
+class DASSample(Container):
+    slot: Slot
+    shard: Shard
+    index: SampleIndex
+    proof: BLSKateProof
+    data: Vector[BLSPoint, POINTS_PER_SAMPLE]
+```
+
+### ShardBlob
+
+The blob of data, effectively a block. Network-only.
+
+```python
+class ShardBlob(Container):
+    # Slot and shard that this blob is intended for
+    slot: Slot
+    shard: Shard
+    # The actual data
+    data: List[BLSPoint, POINTS_PER_SAMPLE * MAX_SAMPLES_PER_BLOCK]
```

Comment:
Why not just be dumb at the cost of redundancy and make the structure have only two params: the `SignedShardHeader` and the `data`?

Reply:
Then we are including both the commitment and the actual data. And then we need to check that these match, by computing the commitment I suppose? So I think we can just drop that information.

That does solve the signature problem though; if the commitment is signed over, and the data matches the commitment, then the data does not have to be signed.

Reply:
BTW, you can give a proof that a commitment is correct that is much easier to verify than recomputing the commitment: You use the hash of the commitment to get a random point (Fiat-Shamir), and then evaluate the polynomial at that point and give a Kate proof that this evaluation is correct. Verifying this proof you only need to evaluate the polynomial at that random point (that's only `n` field operations), and check that the Kate proof is correct, which is much cheaper than `n` group operations.

Also, I like the idea of giving the data in coefficient from in the `ShardBlob`. That can save some computations. Getting the data from it is very cheap, whereas getting the coefficients from the data is a bit more expensive due to all the zero polynomial stuff.